src/js/abstract/base-component.js

```diff
@@ -92,17 +154,19 @@ export default class BaseComponent extends HTMLElement {
     if (this.contextCallback) {
       this._makeContextReady();
     }
-
-    this._isConnected = true;
   }
 
   /**
    * Default behaviour is to re-render on attribute addition, change or removal.
    */
-  attributeChangedCallback() {
-    if (this._isConnected && this._hasRendered) {
-      this.render();
+  attributeChangedCallback(name, newValue, oldValue) {
+    if (ENV !== PROD) {
+      lifecycleLogger(this.logLifecycle)(`+++ attributeChangedCallback -> ${this.nodeName}#${this._id} | ${name} from ${oldValue} to ${newValue}\n`);
     }
+
+    const key = camelize(name);
+
+    this[key] = toProp(newValue);
```

Comment:
@LucaMele 
I would be keen to know, how `removal` is detected. Guess just an empty string `''`.
May we should add something like prop-types but in separate PR, as soon as it becomes an issue. And I would realy like to have automatic tests for this...

Reply:
this is a bit dangerous to be honest.. What if we override a DOM property like style or so on?

Reply:
Absolutely, may we add a blacklist for `['style', `className`, `id`, etc]`which throws?
Or I could a check if the property exists already, then throw 😇

Reply:
yes better with check + throw